### DISCUSSION TITLE
No-op update with empty input

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#184] [Ideas]
I have this kind of boilerplate everywhere:

```ts
const { id, input } = schema.parse(ctx.args)
if (Object.keys(input).length) {
  await db.table.find(id).update(input)
}
```

I was thinking, what if ORM could do the check for me instead, and do a no-op update for an empty object?

Basically, that will mean that an empty update object will generate `select count(*)` instead of `update`, and generate `select *` instead of `update ... returning *`. Is that too magicky? Could it perhaps live in a dedicated method such as `.updateOrSelect`? (I don't like this particular name, just speculating.)

Just in case, currently an empty update simply generates invalid SQL:

```sql
UPDATE "table" SET  WHERE "table"."id" = $1
-- or
UPDATE "table" SET  WHERE "table"."id" = $1 RETURNING *
```

**Comment by romeerez** [comment]
Solved here: https://github.com/romeerez/orchid-orm/issues/185
